collector/src/credential.rs

```diff
@@ -0,0 +1,105 @@
+use crate::Error;
+use hpke_dispatch::{Aead, Kdf, Kem};
+use janus_core::{
+    auth_tokens::{AuthenticationToken, BearerToken},
+    hpke::{self, HpkeKeypair, HpkePrivateKey},
+};
+use janus_messages::{HpkeConfig, HpkeConfigId, HpkePublicKey};
+use serde::{Deserialize, Serialize};
+
+/// Serializable representation of a private collector credential, which contains all secrets a
+/// collector uses to interact with an aggregator. It contains the [`BearerToken`] for
+/// authorization to an aggregator, and the private HPKE configuration for decrypting aggregate
+/// shares.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PrivateCollectorCredential {
+    id: HpkeConfigId,
+    kem: Kem,
+    kdf: Kdf,
+    aead: Aead,
+    public_key: HpkePublicKey,
+    private_key: HpkePrivateKey,
+    // TODO(inahga): Turn this non-optional for the next breaking change cycle.
```

Comment:
process nit: for any TODOs, we generally reference a GH issue rather than a specific person or anything else. (there may still be a few errant TODOs floating around the codebase, but let's avoid introducing any new ones)

Reply:
Yep, I'll be replacing it with an issue number before merging. I didn't do it ahead of time pending the discussion of whether it was necessary or not.